### PR TITLE
[expo-file-system/next] Add options to the create function.

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

It adds two options to the `file.create()` / `directory.create()` call.

It now accepts an object with two optional booleans:

- intermediates – creates all directories in the path if they don't exist
- overwrite – replaces a file/directory with an empty file/directory if it exists already.

# Test Plan

Added test-suite tests.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemDirectory.kt [hunk 1]
```diff
@@ -20,10 +20,21 @@ class FileSystemDirectory(file: File) : FileSystemPath(file) {
     return file.isDirectory
   }
 
-  fun create() {
+  fun create(options: CreateOptions = CreateOptions()) {
```
Comment (lukmccall) [inline packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemDirectory.kt:23]:
```suggestion
  fun create(options: CreateOptions) {
```
You're already passing the default value in the module DSL.

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemFile.kt [hunk 2]
```diff
@@ -29,10 +29,20 @@ class FileSystemFile(file: File) : FileSystemPath(file) {
     return file.isFile
   }
 
-  fun create() {
+  fun create(options: CreateOptions = CreateOptions()) {
```
Comment (lukmccall) [inline packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemFile.kt:32]:
```suggestion
  fun create(options: CreateOptions) {
```

**Review by lukmccall — APPROVED** [review]

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemDirectory.kt (hunk 1 above)
Comment (aleqsio) [reply packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemDirectory.kt:23]: yep, but I'm also using it from other functions so I'd like the default to also exist there

File: apps/test-suite/tests/FileSystemNext.ts [hunk 3]
```diff
@@ -150,13 +150,43 @@ export async function test({ describe, expect, it, ...t }) {
...
+
       it('Creates an empty file', () => {
         const file = new File(testDirectory + 'newFolder');
         file.create();
         expect(file.exists).toBe(true);
         expect(file.text()).toBe('');
       });
 
+      it('Throws an error if the file exists', () => {
+        const file = new File(testDirectory + 'newFolder');
```
Comment (tsapeta) [inline apps/test-suite/tests/FileSystemNext.ts:174]: When are you going to stop doing this? 😛 
```suggestion
        const file = new File(testDirectory, 'newFolder');
```

File: apps/test-suite/tests/FileSystemNext.ts [hunk 4]
```diff
@@ -150,13 +150,43 @@ export async function test({ describe, expect, it, ...t }) {
...
+        file.create();
+        expect(file.exists).toBe(true);
+        expect(file.text()).toBe('');
+      });
+
+      it('Overwrites a file if it exists and `overwrite` is set', () => {
+        const file = new File(testDirectory + 'newFolder');
```
Comment (tsapeta) [inline apps/test-suite/tests/FileSystemNext.ts:181]:
```suggestion
        const file = new File(testDirectory, 'newFolder');
```

File: packages/expo-file-system/ios/Next/FileSystemPath.swift [hunk 5]
```diff
@@ -13,6 +13,20 @@ internal class FileSystemPath: SharedObject {
     try ensurePathPermission(appContext, path: url.path, flag: flag)
   }
 
+  func validateCanCreate(_ options: CreateOptions) throws {
+    if !options.overwrite {
+      guard !(try exists) else {
+        throw UnableToCreateException("it already exists")
+      }
+    }
```
Comment (tsapeta) [inline packages/expo-file-system/ios/Next/FileSystemPath.swift:21]: What a monster 😅 
```suggestion
    if !options.overwrite && try exists {
      throw FileAlreadyExistsException("File already exists")
    }
```

File: packages/expo-file-system/src/next/ExpoFileSystem.types.ts [hunk 6]
```diff
@@ -1,3 +1,14 @@
+type CreateOptions = {
+  /**
+   * Whether to create intermediate directories if they do not exist.
```
Comment (tsapeta) [inline packages/expo-file-system/src/next/ExpoFileSystem.types.ts:3]: Missing `@default`

File: packages/expo-file-system/src/next/ExpoFileSystem.types.ts [hunk 7]
```diff
@@ -1,3 +1,14 @@
...
+   */
+  intermediates?: boolean;
+  /**
+   * Whether to overwrite the file or directory if it exists.
```
Comment (tsapeta) [inline packages/expo-file-system/src/next/ExpoFileSystem.types.ts:7]: Missing `@default`

**Review by tsapeta — APPROVED** [review]

**Pull request merged**